HW3 fastq filtrator

anna-kapitonova opened the pull request.

anna-kapitonova committed:
Update README.md

anna-kapitonova committed:
Create fastq_filtrator.py

anna-kapitonova committed:
Create README.md

anna-kapitonova committed:
Added code of fastq_filtrator

anna-kapitonova committed:
Update fastq_filtrator.py

anna-kapitonova committed:
Update fastq_filtrator.py

anna-kapitonova committed:
Update fastq_filtrator.py

anna-kapitonova committed:
Update fastq_filtrator.py

anna-kapitonova committed:
Update README.md

anna-kapitonova committed:
Update README.md

krglkvrmn commented on HW3_fastq_filtrator/fastq_filtrator.py:
```diff
@@ -0,0 +1,110 @@
+    if type(length_bounds) != tuple:
+        length_bounds = (0, length_bounds)
+    elif type(gc_bounds) != tuple:
+        gc_bounds = (0, gc_bounds)
```
Очень хорошо, что есть обработка передачи одиночного значения, мало кто так сделал

krglkvrmn commented on HW3_fastq_filtrator/fastq_filtrator.py:
```diff
@@ -0,0 +1,110 @@
+            if counter == 4:
+                counter = 0
+                seq = line.strip()
+                gc_content = round(((seq.count('C') + seq.count('G'))/len(seq)*100), 1)  # count gc-content
```
Для универсальности лучше сделать seq.upper()

krglkvrmn commented:
1. @4 и @7 не должны были быть отфильтрованы 
![image](https://user-images.githubusercontent.com/48160953/138569667-dfc852a5-d541-485a-a5d3-a19e65cfbcc1.png)
2. Если передать и `length` и `gc_bounds` числом, то падает с ошибкой (внимательно посмотри на 7 строку)
3. Если качество равно quality_threshold, то мы должны сохранять рид
![image](https://user-images.githubusercontent.com/48160953/138569881-6e6857c2-bcb6-4212-8bd4-aeab0edba755.png)
А так получается, что на дефолтных параметрах всё равно происходит фильтрация, чего быть не должно.

В остальном всё хорошо.

UPD: Видимо в первом пункте тоже дело в quality_threshold

anna-kapitonova committed:
Corrected code according to Roma's comments

Solved the problem with gc_bounds and length_bounds of one-number input. Solved quality-threshold filtration problem.
